napalm/ios/ios.py

```diff
@@ -3760,11 +3763,32 @@ def _get_vlan_all_ports(self, output):
         for line in output:
             vlan_m = find_regexp.match(line)
             if vlan_m:
-                was_vlan_or_cont = True
-                vlan_id = vlan_m.group(1)
-                vlan_name = vlan_m.group(2).strip()
-                interfaces = vlan_m.group(3) or ""
-                vlans[vlan_id] = {"name": vlan_name, "interfaces": []}
+                p_vlan_m = match_pvlan_regexp.match(line)
+                if p_vlan_m:
+                    was_vlan_or_cont = True
+                    vlan_name = vlan_m.group(2)
+                    if (
+                        vlan_name == p_vlan_m.group(2)
+                        and p_vlan_m.group(3) in ["community", "isolated"]
+                        and vlan_name == _vlan_id
+                    ):
+                        vlan_id = p_vlan_m.group(2)
+                    elif vlan_m.group(1) != p_vlan_m.group(2) and p_vlan_m.group(3) in [
+                        "community",
+                        "isolated",
+                    ]:
+                        continue
+                    else:
+                        vlan_id = vlan_m.group(1)
+                    vlan_name = _vlan_name if _vlan_name else vlan_name
```

Comment:
Probably need to add a `strip` call on this line somewhere given the failing test output

Reply:
I gave this a try, and even with a `strip()` the tests would need the following in order to pass with this code:

```diff
diff --git a/test/ios/mocked_data/test_get_vlans/no_show_vlan_all_ports/expected_result.json b/test/ios/mocked_data/test_get_vlans/no_show_vlan_all_ports/expected_result.json
index da9cd8e0..ec258ac5 100644
--- a/test/ios/mocked_data/test_get_vlans/no_show_vlan_all_ports/expected_result.json
+++ b/test/ios/mocked_data/test_get_vlans/no_show_vlan_all_ports/expected_result.json
@@ -227,15 +227,13 @@
   "2432": {
     "name": "Vlan2432",
     "interfaces": [
-      "GigabitEthernet0/1",
-      "GigabitEthernet0/2"
+      "Port-channel1"
     ]
   },
   "2433": {
     "name": "Vlan2433",
     "interfaces": [
-      "GigabitEthernet0/3",
-      "GigabitEthernet0/4"
+      "Port-channel1"
     ]
   }
 }
```

So I believe there's more work required in this PR, re-scheduling for the next release.

Reply:
@mirceaulinic @bewing I figured it'd need more work, which is why I changed it back to draft. I'm just back from PTO, will take a look at this coming week.

Reply:
Cool - thanks for the heads up @vvas1lev!